sklearn/utils/sparsefuncs_fast.pyx

```diff
@@ -27,27 +27,35 @@ def csr_row_norms(X):
     """L2 norm of each row in CSR matrix X."""
     if X.dtype not in [np.float32, np.float64]:
         X = X.astype(np.float64)
-    return _csr_row_norms(X.data, X.shape, X.indices, X.indptr)
+    return np.asarray(_csr_row_norms(X.data, X.indices, X.indptr))
 
 
-def _csr_row_norms(cnp.ndarray[floating, ndim=1, mode="c"] X_data,
-                   shape,
-                   cnp.ndarray[integral, ndim=1, mode="c"] X_indices,
-                   cnp.ndarray[integral, ndim=1, mode="c"] X_indptr):
+def _csr_row_norms(
+    floating[::1] X_data,
+    integral[::1] X_indices,
+    integral[::1] X_indptr,
+):
     cdef:
-        unsigned long long n_samples = shape[0]
-        unsigned long long i
+        integral n_samples = X_indptr.shape[0] - 1
+        integral i
```

Comment:
I am not sure about this change. If `integral` is 32 bits then, we will not support for large CSR matrix (cf. https://github.com/scikit-learn/scikit-learn/pull/9663)

Reply:
This change works properly: this males `n_samples` and `i` use the same type as the one used for X's `indices` and `indptr` (hence, in case large CSR matrix, `int64` will be used for all of them).